src/domain/transaction/stellar/stellar_transaction.rs

```diff
@@ -171,6 +176,166 @@ where
+            .transaction_repository()
+            .partial_update(tx_id, update_req)
+            .await?;
+
+        self.send_transaction_update_notification(&updated_tx)
+            .await?;
+        Ok(updated_tx)
+    }
+
+    /// Helper function to re-queue a transaction status check job.
+    async fn requeue_status_check(
+        &self,
+        tx: &TransactionRepoModel,
+    ) -> Result<(), TransactionError> {
+        self.job_producer()
+            .produce_check_transaction_status_job(
+                TransactionStatusCheck::new(tx.id.clone(), tx.relayer_id.clone()),
+                Some(STELLAR_DEFAULT_STATUS_RETRY_DELAY_SECONDS),
+            )
+            .await?;
+        Ok(())
+    }
+
+    /// Parses the transaction hash from the network data and validates it.
+    /// Returns a `TransactionError::ValidationError` if the hash is missing, empty, or invalid.
+    fn parse_and_validate_hash(&self, tx: &TransactionRepoModel) -> Result<Hash, TransactionError> {
+        let stellar_network_data = tx.network_data.get_stellar_transaction_data()?;
+
+        let tx_hash_str = stellar_network_data.hash.as_deref().filter(|s| !s.is_empty()).ok_or_else(|| {
+            TransactionError::ValidationError(format!(
+                "Stellar transaction {} is missing or has an empty on-chain hash in network_data. Cannot check status.",
+                tx.id
+            ))
+        })?;
+
+        let stellar_hash: Hash = tx_hash_str.parse().map_err(|e: Error| {
+            TransactionError::UnexpectedError(format!(
+                "Failed to parse transaction hash '{}' for tx {}: {:?}. This hash may be corrupted or not a valid Stellar hash.",
+                tx_hash_str, tx.id, e
+            ))
+        })?;
+
+        Ok(stellar_hash)
+    }
+
+    /// Handles the logic when a Stellar transaction is confirmed successfully.
+    async fn handle_stellar_success(
+        &self,
+        tx: TransactionRepoModel,
+        _provider_response: soroban_rs::stellar_rpc_client::GetTransactionResponse, // May be needed later for more details
+    ) -> Result<TransactionRepoModel, TransactionError> {
+        let confirmed_tx = self
+            .finalize_transaction_state(
+                tx.id.clone(),
+                TransactionStatus::Confirmed,
+                None,
+                Some(Utc::now().to_rfc3339()),
+            )
+            .await?;
+
+        if let Some(oldest_pending) = self
+            .transaction_repository()
+            .find_oldest_pending_for_relayer(&confirmed_tx.relayer_id)
+            .await?
+        {
+            info!(
+                "Enqueuing next pending transaction {} for relayer {}",
+                oldest_pending.id, confirmed_tx.relayer_id
+            );
+            self.enqueue_submit(&oldest_pending, None).await?;
+        }
+        Ok(confirmed_tx)
+    }
+
+    /// Handles the logic when a Stellar transaction has failed.
+    async fn handle_stellar_failed(
```

Comment:
So if tx fails and there are others in pending state we do not enqueue next one.

I guess that will make relayer stuck.

For stellar, when tx fails is that sequence number used or we need to handle it to make relayer working?

For example let's use case when we have txs in pending state with assigned higher sequence numbers.